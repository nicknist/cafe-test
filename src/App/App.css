@import url('https://fonts.googleapis.com/css?family=Pacifico');

.App {
  display: flex;
  flex-direction: column;
  align-items: center;
  background-image: url('../assets/cafe-img.jpeg');
  width: 100%;
  height: 100vh;
  background-repeat: no-repeat;
  background-size: cover;
  text-align: center;
}

.app-title {
  color: teal;
  font-family: 'Pacifico', cursive;
  font-size: 50px;
}

.single-res {
  color: teal;
  background-color: white;
  border: 2px solid teal;
  margin: 15px;
  padding: 10px;
}

.resy-container {
  display: flex;
  flex-flow: wrap;
}

button {
  background-color: teal;
  color: white;
}

input {
  margin: 0px 10px;
}
