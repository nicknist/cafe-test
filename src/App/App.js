import React, { Component } from 'react';
import './App.css';
import DisplayContainer from '../DisplayContainer/DisplayContainer';
import MakeReservationForm from '../MakeReservationForm/MakeReservationForm';

class App extends Component {
  constructor() {
    super();
    this.state = {
      reservations: [],
      idNumber: 10
    }
  }

  addReservation = (newReservation) => {
    let newRes = {...newReservation}
    newRes.idNumber = this.state.idNumber;
    let newID = this.state.idNumber + 1
    this.setState({
      reservations: [...this.state.reservations, newRes],
      idNumber: newID
    })
  }

  componentDidMount() {
    fetch('http://localhost:3001/api/v1/reservations')
        .then(res => res.json())
        .then(data => this.setState({ reservations: data }))
        .catch(err => console.log(err))
  }

  render() {
    return (
      <div className="App">
        <header>
          <h1 className='app-title'>Turing Cafe Reservations</h1>
        </header>
        <main>
          <div className='resy-form'>
            <MakeReservationForm addReservation={this.addReservation}/>
          </div>
          <DisplayContainer className='resy-container' reservations={this.state.reservations}/>
        </main>
      </div>
    )
  }
}

export default App;
