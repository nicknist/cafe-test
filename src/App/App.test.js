import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<App />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})
