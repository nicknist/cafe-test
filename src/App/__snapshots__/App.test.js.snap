// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<div
  className="App"
>
  <header>
    <h1
      className="app-title"
    >
      Turing Cafe Reservations
    </h1>
  </header>
  <main>
    <div
      className="resy-form"
    >
      <MakeReservationForm
        addReservation={[Function]}
      />
    </div>
    <DisplayContainer
      className="resy-container"
      reservations={Array []}
    />
  </main>
</div>
`;
